detector_ros/src/detector_ros.cpp

```diff
@@ -0,0 +1,63 @@
+#include <detector_ros/detector_ros.hpp>
+
+namespace ariitk::detector_ros {
+
+int h_min,s_min,v_min,h_max,s_max,v_max,canny_lower,canny_upper,min_contour_area,canny_ker;
+
+void DetectorROS::init(ros::NodeHandle& nh) {
+    
+    img_sub_ = nh.subscribe("image_raw", 1, &DetectorROS::imageCallback, this);
+
+    nh.getParam("h_min",h_min);nh.getParam("s_min",s_min);nh.getParam("v_min",v_min);
+    nh.getParam("h_max",h_max);nh.getParam("s_max",s_max);nh.getParam("v_max",v_max);
+    nh.getParam("canny_lower",canny_lower);nh.getParam("canny_upper",canny_upper);nh.getParam("min_contour_area",min_contour_area);nh.getParam("canny_ker",canny_ker);
+    
+    ros::NodeHandle nh_private("~");
```

Comment:
also use private node handle to get the params, since you'll load the param file inside the node